module-system/module-implementations/sov-blob-storage/tests/capability_tests.rs

```diff
@@ -120,16 +151,251 @@ fn priority_sequencer_flow_general() {
         )
         .unwrap();
 
+        // Run any extra logic provided by the test for this slot
+        if let Some(next_slot_info) = test_info.peek() {
+            if next_slot_info.slot_number == slot_number {
+                let next_slot_info = test_info.next().unwrap();
+                // If applicable, assert that the expected number of blobs was processed
+                if let Some(expected) = next_slot_info.expected_blobs_to_process {
+                    assert_eq!(expected, blobs_to_execute.len())
+                }
+
+                // If applicable, send the requested callmessage to the blob_storage module
+                if let Some((msg, sender)) = next_slot_info.early_processing_request_with_sender {
+                    runtime
+                        .blob_storage
+                        .call(msg, &DefaultContext::new(sender), &mut working_set)
+                        .unwrap();
+                    has_processed_blobs_early = true;
+                }
+            }
+        }
+
+        // Check that the computed list of blobs is the one we expected
         for blob in blobs_to_execute {
-            let expected = expected_blobs.next().unwrap();
-            assert!(expected.should_get_processed_in() == slot_number);
-            assert_blobs_are_equal(expected.blob, blob, &format!("slot {:?}", slot_number));
+            let expected: BlobWithAppearance<MockBlob> = expected_blobs.next().unwrap();
+            if !has_processed_blobs_early {
+                assert_eq!(expected.must_be_processed_by(), slot_number);
+            }
+            assert_blobs_are_equal(expected.blob, blob, &format!("Slot {}", slot_number));
         }
     }
     // Ensure that all blobs have been processed
     assert!(expected_blobs.next().is_none());
 }
 
+#[test]
+fn bonus_blobs_are_delivered_on_request() {
+    // If blobs are deferred for less than two slots, "early processing" is not possible
+    if DEFERRED_SLOTS_COUNT < 2 {
```

Comment:
Do you think it makes sense to have a test with this condition, to show how rollup suppose to behave?

My understanding is, that if `DEFERRED_SLOTS_COUNT < 2` logic suppose to be the same as with flushing request, as without it. 

It might sound stupid, but having test prove that would be nice, in case if there are some subtle bugs.

Reply:
Good idea. Added tests and had to special case some things when `DEFERRED_SLOTS_COUNT == 0`. I don't think that's a case we'll likely run into, but this way it's covered.